pkg/kubelet/kubelet.go

```diff
@@ -927,41 +926,17 @@ func (kl *Kubelet) pullImage(img string, ref *api.ObjectReference) error {
 
 // Kill all running containers in a pod (includes the pod infra container).
 func (kl *Kubelet) killPod(pod kubecontainer.Pod) error {
-	// Send the kills in parallel since they may take a long time.
-	errs := make(chan error, len(pod.Containers))
-	wg := sync.WaitGroup{}
-	for _, container := range pod.Containers {
-		wg.Add(1)
-		go func(container *kubecontainer.Container) {
-			defer util.HandleCrash()
-			// Call the networking plugin for teardown.
-			// TODO: Handle this without signaling the pod infra container to
-			// adapt to the generic container runtime.
-			if container.Name == dockertools.PodInfraContainerName {
-				err := kl.networkPlugin.TearDownPod(pod.Namespace, pod.Name, dockertools.DockerID(container.ID))
-				if err != nil {
-					glog.Errorf("Failed tearing down the infra container: %v", err)
-					errs <- err
-				}
-			}
-			err := kl.containerManager.KillContainer(container.ID)
-			if err != nil {
-				glog.Errorf("Failed to delete container: %v; Skipping pod %q", err, pod.ID)
-				errs <- err
-			}
-			wg.Done()
-		}(container)
-	}
-	wg.Wait()
-	close(errs)
-	if len(errs) > 0 {
-		errList := []error{}
-		for err := range errs {
-			errList = append(errList, err)
+	// TODO(vmarmol): Consider handling non-Docker runtimes, the plugins are not friendly to it today.
+	container, err := kl.containerManager.GetPodInfraContainer(pod)
```

Comment:
How badly we inject networkPlugin to ContainerRuntime, so that PodInfraContainer can be completely handled by ContainerRuntime, such as docker runtime here?